Wrappers/Python/test/test_viewer_main_windows.py

```diff
@@ -31,30 +31,55 @@ class TestViewerMainWindow(unittest.TestCase):
     '''
 
     def setUp(self):
+        pass
+        
+    def tearDown(self) -> None:
+        pass
+        
+    def test_all(self):
         global _instance
         if _instance is None:
             _instance = QApplication(sys.argv)
 
-    def test_init(self):
+        # https://stackoverflow.com/questions/5387299/python-unittest-testcase-execution-order
+        # https://stackoverflow.com/questions/11145583/unit-and-functional-testing-a-pyside-based-application
+        self._test_init()
+        self._test_init_calls_super_init()
+        self._test_createAppSettingsDialog_calls_setAppSettingsDialogWidgets()
+        self._test_acceptViewerSettings_when_gpu_checked()
+        self._test_acceptViewerSettings_when_gpu_unchecked()
+        self._test_setDefaultDownsampledSize()
+        self._test_getDefaultDownsampledSize()
+        self._test_getTargetImageSize_when_vis_size_is_None()
+        self._test_getTargetImageSize_when_vis_size_is_not_None()
+        self._test_updateViewerCoords_with_display_unsampled_coords_selected()
+        self._test_updateViewerCoords_with_display_downsampled_coords_selected()
+        self._test_updateViewerCoords_with_3D_viewer()
+        self._test_updateViewerCoords_with_no_img3D()   
+
+        del _instance
+
+
+    def _test_init(self):
         vmw = ViewerMainWindow(title="Testing Title", app_name="testing app name")
         assert vmw is not None
 
-    def test_init_calls_super_init(self):
+    def _test_init_calls_super_init(self):
         vmw = ViewerMainWindow(title="Testing Title", app_name="testing app name")
         # If the super init is called, then the following attributes should be set:
         assert vmw.app_name == "testing app name"
         assert vmw.threadpool is not None
         assert vmw.progress_windows == {}
 
-    def test_createAppSettingsDialog_calls_setAppSettingsDialogWidgets(self):
+    def _test_createAppSettingsDialog_calls_setAppSettingsDialogWidgets(self):
```

Comment:
What's the reason for the underscores in front of the methods?

Reply:
https://github.com/vais-ral/CILViewer/issues/367 You can't run more than one QApplication so I thought to run the tests sequentially, which means not run them as indipendent tests.
Unfortunately this works to a certain extent, because you can't run 2 different test classes because they might run at the same time.